gestalt/__init__.py

```diff
@@ -155,6 +154,58 @@ def build_config(self) -> None:
         self.__conf_data = self.__flatten(self.__conf_data,
                                           sep=self.__delim_char)
 
+        self.__parse_keys()
+        self.__interpolate_keys()
+
+    def __parse_keys(self) -> None:
+        """Parses the keys in the configuration data.
+
+        Raises:
+            RuntimeError: If the configuration data is not valid
+        """
+        for k, v in self.__conf_data.items():
+            if not isinstance(v, str):
+                continue
+            m = self.regex_pattern.search(v)
+            if m is None:
+                continue
+            if m.group(1) not in self.__providers:
+                raise RuntimeError(
+                    "Provider not configured yet expect to be used")
+            if v in self.__secret_map:
```

Comment:
Just want to confirm you want the entire non-interpolated string value here.

Reply:
I think taken care off 😄